Source/Core/DolphinQt/GameList/GameListModel.cpp

```diff
@@ -208,36 +210,11 @@ QVariant GameListModel::headerData(int section, Qt::Orientation orientation, int
   if (orientation == Qt::Vertical || role != Qt::DisplayRole)
     return QVariant();
 
-  switch (static_cast<Column>(section))
-  {
-  case Column::Title:
-    return tr("Title");
-  case Column::ID:
-    return tr("ID");
-  case Column::Banner:
-    return tr("Banner");
-  case Column::Description:
-    return tr("Description");
-  case Column::Maker:
-    return tr("Maker");
-  case Column::FileName:
-    return tr("File Name");
-  case Column::FilePath:
-    return tr("File Path");
-  case Column::Size:
-    return tr("Size");
-  case Column::FileFormat:
-    return tr("File Format");
-  case Column::BlockSize:
-    return tr("Block Size");
-  case Column::Compression:
-    return tr("Compression");
-  case Column::Tags:
-    return tr("Tags");
-  default:
-    break;
-  }
-  return QVariant();
+  static const std::array<QString, static_cast<std::size_t>(Column::Count)> headers = {
```

Comment:
Maybe add a comment this'll need to be non-const if we ever support changing the language at runtime without restarting dolphin (we really should, but alas), otherwise the translation will only use the proper string once.

Looks good otherwise.

Reply:
That would make this change really inefficient relative to the current switch-case (constructing an array of strings just to pick one). If this must be compliant to a possible future which supports runtime language changes, then I think I wouldn't change this at all.

Reply:
You can also use a static array of raw strings like:

```cpp
static constexpr std::array headers = {
  QT_TR_NOOP("A"),
  QT_TR_NOOP("B"),
  // ... etc
}; 
return tr(headers[section]);
```

which gets you the best of both worlds and also makes it work out of the box when runtime language switching is a thing.

Reply:
Now that's something I can get behind. It even lets me make the std::array constexpr!